File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03213/p03213_num7_parsed.py
import math

def count_shichi_go_numbers(n):
    factors = [0] * (n + 1)
    
    # Calculate the prime factors of N!
    for i in range(2, n + 1):
        while n % i == 0:
            factors[i] += 1
            n //= i

    count = 0

    # Calculate the number of Shichi-Go numbers among the divisors of N!
    for factor_count in factors:
        if factor_count >= 74:
            count += 1

    return count

# Get the input
n = int(input())

# Count the Shichi-Go numbers
shichi_go_count = count_shichi_go_numbers(n)

# Print the result
print(shichi_go_count)

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03213/p03213_num7_parsed.cpp
#include <iostream>
#include <vector>
using namespace std;

// Function to calculate the prime factors of a number
vector<int> primeFactors(int n) {
    vector<int> factors;
    for (int i = 2; i <= n; i++) {
        while (n % i == 0) {
            factors.push_back(i);
            n = n / i;
        }
    }
    return factors;
}

// Function to calculate the number of divisors of a number
int calculateDivisors(vector<int> factors) {
    vector<int> primes;
    int count = 1;
    
    // Calculate the frequency of each prime factor
    for (int i = 0; i < factors.size(); i++) {
        if (find(primes.begin(), primes.end(), factors[i]) == primes.end()) {
            int freq = count(factors.begin(), factors.end(), factors[i]);
            primes.push_back(factors[i]);
            count *= (freq + 1);
        }
    }
    
    return count;
}

int main() {
    int N;
    cin >> N;
    
    // Calculate the prime factors of N!
    vector<int> factors = primeFactors(N);
    
    // Calculate the number of Shichi-Go numbers
    int shichiGoCount = 0;
    for (int i = 1; i <= N; i++) {
        vector<int> numFactors = primeFactors(i);
        int divisorCount = calculateDivisors(numFactors);
        
        if (divisorCount == 75) {
            shichiGoCount++;
        }
    }
    
    // Output the result
    cout << shichiGoCount << endl;
    
    return 0;
}